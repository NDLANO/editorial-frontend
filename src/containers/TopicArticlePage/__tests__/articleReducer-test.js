/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import reducer, { actions } from '../articleDucks';

test('reducers/articles initalState', () => {
  const nextState = reducer(undefined, { type: 'Noop' });

  expect(nextState).toMatchSnapshot();
});

test('reducers/articles set article', () => {
  const nextState = reducer(undefined, { type: actions.setArticle, payload: { id: 1, title: 'Unit test' } });

  expect(nextState).toMatchSnapshot();
});

test('reducers/articles set multiple articles', () => {
  const state = reducer(undefined, { type: actions.setArticle, payload: { id: 1, title: 'Unit test 1' } });
  const nextState = reducer(state, { type: actions.setArticle, payload: { id: 2, title: 'Unit test 2' } });

  expect(nextState).toMatchSnapshot();
});

test('reducers/article overwrite articles with same id', () => {
  const nextState = reducer({
    1: { id: 1, title: 'Unit test 1' },
  }, { type: actions.setArticle, payload: { id: 1, title: 'Unit test 2' } });

  expect(nextState).toMatchSnapshot();
});
