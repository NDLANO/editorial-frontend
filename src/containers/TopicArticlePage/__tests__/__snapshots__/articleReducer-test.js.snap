// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/article overwrite articles with same id 1`] = `
Object {
  "1": Object {
    "id": 1,
    "title": "Unit test 1",
  },
  "all": Object {
    "1": Object {
      "id": 1,
      "title": "Unit test 2",
    },
  },
}
`;

exports[`reducers/articles initalState 1`] = `
Object {
  "all": Object {},
  "isSaving": false,
}
`;

exports[`reducers/articles set article 1`] = `
Object {
  "all": Object {
    "1": Object {
      "id": 1,
      "title": "Unit test",
    },
  },
  "isSaving": false,
}
`;

exports[`reducers/articles set multiple articles 1`] = `
Object {
  "all": Object {
    "1": Object {
      "id": 1,
      "title": "Unit test 1",
    },
    "2": Object {
      "id": 2,
      "title": "Unit test 2",
    },
  },
  "isSaving": false,
}
`;
